website/static/js/citationList.js

```diff
@@ -21,31 +23,117 @@ var formatCitation = function(style, data, format) {
+     constructor: function() {
+        var self = this;
+        self.loading = ko.observable(true);
+        self.customCitationText = ko.observable('');
+        self.initialCustomCitationText = ko.observable();
+        self.apa = ko.observable();
+        self.mla = ko.observable();
+        self.chicago = ko.observable();
+        self.showEdit =  ko.observable(false);
+        self.page = ko.computed(function () {
+            if(self.loading()){
+                return 'loading';
+            } else if(self.showEdit()) {
+                return 'edit';
+            } else if(self.customCitationText() === '') {
+                return 'standard';
+            } else {
+                return 'custom';
+            }
+        });
+        self.nodeType = ko.computed(function () {
+            return ctx.node.isRegistration ? 'registrations' : 'nodes';
+        });
 
-    self.apa = ko.observable();
-    self.mla = ko.observable();
-    self.chicago = ko.observable();
-};
+        makeClient($('#custom-citation-copy-button')[0]);
+    },
+    showEditBox: function() {
+        var self = this;
+        self.showEdit(true);
+        self.initialCustomCitationText(self.customCitationText());
+    },
+    cancelCitation: function() {
+        var self = this;
+        self.showEdit(false);
+        self.customCitationText(self.initialCustomCitationText());
+    },
+    saveCitation: function() {
+        var self = this;
+        self.loading(true);
+        self.updateCustomCitation();
+    },
+    clearCitation: function() {
+        var self = this;
+        if(self.customCitationText() !== '') {
+            self.customCitationText('');
+            self.loading(true);
+            self.updateCustomCitation();
+        } else {
+            self.showEdit(false);
+        }
+    },
+    updateCustomCitation: function() {
+        var self = this;
+        var payload = {
+            'data': {
+                'id' : ctx.node.id,
+                'type' :  self.nodeType(),
+                'attributes' : {
+                    'title': ctx.node.title,
+                    'category': ctx.node.category,
+                    'custom_citation_text': self.customCitationText()
+                 }
+            }
+        };
+        self.showEdit(false);
+        $.ajax({
+            url: ctx.apiV2Prefix  + self.nodeType() + '/' + ctx.node.id + '/',
+            type: 'PUT',
+            dataType: 'json',
+            contentType: 'application/json',
+            beforeSend: $osf.setXHRAuthorization,
+            data: JSON.stringify(payload)
+        }).done(function(data) {
+            self.fetch();
+        }).fail(function() {
+            $osf.growl('Error', 'Your custom citation not updated. Please refresh the page and try ' +
+            'again or contact ' + $osf.osfSupportLink() + ' if the problem persists.', 'danger');
+        });
+    },
+    fetch: function() {
+        var self = this;
+        $.ajax({
```

Comment:
Any reason to not use our [`ajaxJSON`](https://github.com/CenterForOpenScience/osf.io/blob/0.153.5/website/static/js/osfHelpers.js#L68) helper here?